### DISCUSSION TITLE
why can not query when datacoord failure

### DISCUSSION THREAD

**Discussion by zhangchao** [milvus-io/milvus#25636] [Q&A and General discussion]
milvus 2.2.8 
we found query failure when datacoord node failure, it looks strange behavior.From my understanding , query node only depends on proxy, querycoord and pulsar  . why it depends on datacoord or indexcoord 
`│ querycoord [2023/07/17 03:25:27.936 +00:00] [ERROR] [grpcclient/client.go:330]  [role=datacoord] [error="err: find no available datacoord, check datacoord state\n, 
/go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 
github.com/milvus-io/milvus/internal/util/trace.StackTrace\n/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:329 
github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).ReCall\n/go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/client/client.go:435 
github.com/milvus-io/milvus/internal/distributed/datacoord/client.(*Client).GetRecoveryInfoV2\n/go/src/github.com/milvus-io/milvus/internal/querycoordv2/meta/coordinator_broker.go:153 `

**Comment by xiaofan-luan** [comment]
query coord depends on data and index also because it has to get data.

You will need to figure out why datacoord is not working as expected

**Reply by zhangchao** [reply]
in fact, I don't understand what the data depends on.  It looks not good design, when "write path" fail, it will impact the "query path"

**Comment by xiaofan-luan** [comment]
> in fact, I don't understand what the data depends on. It looks not good design, when "write path" fail, it will impact the "query path"

data can not be load if datacoord crashed.
And if you use eventual consistency, you can still search while datacoord crash

**Reply by xiaofan-luan** [reply]
As you list, GetRecoveryInfoV2 is try to update segment info. even with the failure you search could still work
